org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/internal/project/registry/MemoryConsumptionTest.java

```diff
@@ -69,7 +69,10 @@ public void testImportLongBuildChain() throws Exception {
       waitForJobsToComplete(monitor);
       for(IProject p : ResourcesPlugin.getWorkspace().getRoot().getProjects()) {
         if(p.hasNature(IMavenConstants.NATURE_ID)) {
-          poms.remove(p.getFile("pom.xml").getLocation().toFile());
+          File pomFile = p.getFile("pom.xml").getLocation().toFile();
+          if (!poms.remove(pomFile)) {
+            Assert.fail("Could not remove pom path " + pomFile + " from list of projects to be imported: " + poms);
```

Comment:
Couldn't you just use `Assert.assertTrue()`?

Reply:
Makes sense.

Reply:
Done in https://github.com/eclipse-m2e/m2e-core/pull/1154/commits/562182c8eb2bcdcdcf719012dd79c3aced3a69fd.